server.js

```diff
@@ -40,18 +40,16 @@ defaultClient.basePath = 'https://connect.squareupsandbox.com';
 app.post('/process-payment', async (req, res) => {
   const request_params = req.body;
 
-  // length of idempotency_key should be less than 45
-  const idempotency_key = crypto.randomBytes(22).toString('hex');
-
   // Charge the customer's card
   const payments_api = new squareConnect.PaymentsApi();
   const request_body = {
     source_id: request_params.nonce,
+    location_id: request_params.location_id,
     amount_money: {
       amount: 100, // $1.00 charge
       currency: 'USD'
     },
-    idempotency_key: idempotency_key
+    idempotency_key: request_params.idempotency_key;
```

Comment:
May I know why we are making this change?

Reply:
During our documentation review for Developer Success Summary, we identified an issue with duplicate payments. Specifically, new developers don't understand how to properly use idempotency key, so this change is an attempt to better educate new developers on how to use the idempotency key on the client side, to prevent duplicate charges (ie only create a new idempotency key when necessary). on the front end, there will also be the change of disabling the payment button to help with this scenario. Feel free to add anything else, John.

Reply:
That accurately captures the issue.  Leaving the idempotency key generation in server.js, if the client POSTS the same nonce multiple times, the server side generates a new key with each request, making each payment request appear to be unique (but should not be) - despite the fact that the same nonce is used.

Reply:
That makes sense. I'm OK with this change.

We need keep a balance between simplicity and the functionality. The walkthrough is not set to show developer all the best practices.